### PR TITLE
Consul DNS Package

### LINKED ISSUE #1230: DNSMasq Configuration blocks metadata.google.internal
- Ansible version (`ansible --version`): 1.9.2
- Python version (`python --version`): 2.7.9
- Git commit hash or branch: feature/fewer-logs-please
- Cloud Environment: GCE
- Terraform version (`terraform version`): 0.6.12

Consul logs get a lot of these warnings on GCE:

```
Mar 04 15:23:03 asteroid-control-01 consul[7194]: 2016/03/04 15:23:03 [WARN] dns: QName invalid: metadata.google.internal.
```

(and I do mean a **lot**, periods of 30 seconds producing 11ish logs per second)

It looks like this is because we're modifying the default `resolv.conf` and overriding information:

```
search consul c.asteris-mi.internal. google.internal. c.asteris-mi.internal
nameserver 127.0.0.1
nameserver 169.254.169.254
nameserver 10.0.0.1
```

This means that while `metadata.google.internal` should be queried from `169.254.169.254` (it works with `dig @169.254.169.254 metadata.google.internal`), it's being queried from DNSMasq. DNSMasq asks Consul, which doesn't know about it, and then DNSMasq is configured to ask Google's public DNS (`8.8.8.8` and `8.8.4.4`).

We can probably avoid this error by moving the system `resolv.conf` to `resolv.cloud.conf` and configuring DNSMasq to use it as a fallback instead of public DNS. This work probably blocks completely internal deployments, as well.

### PR DESCRIPTION
**Pull request by BrianHicks**
- [x] Installs cleanly on a fresh build of most recent master branch
- [x] Upgrades cleanly from the most recent release
- [x] Updates documentation relevant to the changes
- [x] Rebases cleanly onto the latest master

Fixes #1230

### CODE REVIEW

**Comment by ryane** [comment]
provisioned on top of 1.0.3 rc1 AWS cluster. installed cleanly, no problems so far

**Comment by BrianHicks** [comment]
Looks like the build failed, but because Docker hub timed out on one cloud. The others worked fine. Should we consider that a green or retry it? @sehqlr

**Comment by sehqlr** [comment]
I would say retry it once.

**Comment by BrianHicks** [comment]
Right, I'll rebase and then we can get this merged! :thumbsup:

**Comment by BrianHicks** [comment]
As of right now, it would be a fast-forward merge. So we're good here.

File: roles/dnsmasq/tasks/main.yml [hunk 1]
```diff
@@ -1,60 +1,26 @@
...
-    - dnsmasq
-
-- name: ensure dnsmasq.d directory exists
+- name: remove consul from list of search domains
   sudo: yes
-  file:
-    path: /etc/NetworkManager/dnsmasq.d
-    state: directory
+  command: sed -i'' 's/consul *//g' /etc/resolv.conf.masq
+  register: resolve_without_consul
```
Comment (ryane) [inline roles/dnsmasq/tasks/main.yml]: is this used for anything?

Comment (BrianHicks) [inline roles/dnsmasq/tasks/main.yml]: just an old line I forgot to remove. Thanks for catching it!

**Comment by ryane** [comment]
lgtm, successfully tested on aws. automated digitalocean build failed due to unrelated problem

**Pull request merged**